src/aks-preview/azext_aks_preview/aks_draft/commands.py

```diff
@@ -0,0 +1,300 @@
+                                resource_group=resource_group,
+                                provider=provider,
+                                gh_repo=gh_repo)
+
+    run_successful = _run(file_path, 'setup-gh', setup_gh_args)
+    if not run_successful:
+        raise ValueError('`az aks draft setup-gh` was NOT executed successfully')
+
+    generate_workflow_args = _build_args(cluster_name=cluster_name,
+                                         registry_name=registry_name,
+                                         container_name=container_name,
+                                         resource_group=resource_group,
+                                         destination=destination,
+                                         branch=branch)
+    run_successful = _run(file_path, 'generate-workflow', generate_workflow_args)
+    if run_successful:
+        _run_finish()
+    else:
+        raise ValueError('`az aks draft generate-workflow` was NOT executed successfully')
+
+
+# `az aks draft update` function
+def aks_draft_cmd_update(host: str, certificate: str, destination: str, download_binary: bool) -> None:
+    file_path, arguments = _pre_run(download_binary, host=host, certificate=certificate, destination=destination)
+    run_successful = _run(file_path, 'update', arguments)
+    if run_successful:
+        _run_finish()
+    else:
+        raise ValueError('`az aks draft update` was NOT executed successfully')
+
+
+# Returns binary file path and arguments
+def _pre_run(download_binary: bool, **kwargs) -> Tuple[str, List[str]]:
+    file_path = _binary_pre_check(download_binary)
+    if not file_path:
+        raise ValueError('Binary check was NOT executed successfully')
+    arguments = _build_args(kwargs)
+    return file_path, arguments
+
+
+# Executes the Draft command
+# Returns True if the process executed sucessfully, False otherwise
+def _run(binary_path: str, command: str, arguments: List[str]) -> bool:
+    if binary_path is None:
+        raise ValueError('The given Binary path was null or empty')
+
+    logging.info(f'Running `az aks draft {command}`')
+    cmd = [binary_path, command] + arguments
+    process = subprocess.Popen(cmd)
+    exit_code = process.wait()
+    return exit_code == 0
+
+
+# Function for clean up logic
+def _run_finish():
+    # Clean up logic can go here if needed
+    logging.info('Finished running Draft command')
+
+
+def _build_args(args_dict: Dict[str, str] = None, **kwargs) -> List[str]:
+    if not args_dict:
+        args_dict = kwargs
+    args_list = []
+    for key, val in args_dict.items():
+        arg = key.replace('_', '-')
+        if val:
+            args_list.append(f'--{arg}={val}')
+    return args_list
+
+
+# Returns the path to Draft binary. None if missing the required binary
+def _binary_pre_check(download_binary: bool) -> Optional[str]:
+    logging.info('The Draft binary check is in progress...')
+    draft_binary_path = _get_existing_path()
+
+    if download_binary:
```

Comment:
💡Sharing a **quick observation\thoughts** : Please see line above: `draft_binary_path` so even if the binary will exist, we go ahead and download the binary? i.e. does exist makes any change to the decision tree of `download_binary` ?

Reply:
Thank you for this Tats!